sympy/core/numbers.py

```diff
@@ -155,25 +155,30 @@ def igcd(*args):
     5
 
     """
-    a = args[0]
-    for b in args[1:]:
-        try:
-            a = _gcdcache[(a, b)]
-        except KeyError:
-            a, b = as_int(a), as_int(b)
-
-            if a and b:
+    a = abs(as_int(args[0])) if args else 0
+    k = 1
+    if a != 1:
+        while k < len(args):
+            b = args[k]
+            k += 1
+            try:
+                a = _gcdcache[(a, b)]
+            except KeyError:
+                b = as_int(b)
+                if not b:
+                    continue
+                if b == 1:
+                    a = 1
+                    break
                 if b < 0:
                     b = -b
-
+                t = a, b
                 while b:
                     a, b = b, a % b
-            else:
-                a = abs(a or b)
-
-            _gcdcache[(a, b)] = a
-        if a == 1 or b == 1:
-            return 1
+                _gcdcache[t] = _gcdcache[t[1], t[0]] = a
```

Comment:
Instead of `_gcdcache[t] = _gcdcache[t[1], t[0]] = a`, you can simply do `_gcdcache[t] = a`